.idea/encodings.xml

```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
all these ``.idea`` files are, I think, mac OS thingy's and we don't want them in the repo.

We can add .idea to the .gitignore file in this repo so this doesn't happen. We have two options here....Yao could remove these from the git repo using 
``git rm .idea/*``  (or sthg like that), or we could kill this PR and add the current code to a new PR without merging this one.

OK, Googling around a bit, it seems that the best if if you add ``.idea`` to your global gitignore:
``~/.gitignore_global``
e.g., 
```
cd ~        # this takes you to your personal root directory
ls .gitignore_global    # if there is one there you will see it.  If you see it, open it in a text editor to edit it.  If you don't see it....
touch .gitignore   ## will create it.  Then open it in a text editor and type in the file

# file: ~/.gitignore_global
.DS_Store
.idea

the save it.
```